src/libraries/System.Linq/src/System/Linq/Count.cs

```diff
@@ -20,10 +20,12 @@ public static int Count<TSource>(this IEnumerable<TSource> source)
                 return collectionoft.Count;
             }
 
-            if (source is IIListProvider<TSource> listProv)
+#if !OPTIMIZE_FOR_SIZE
+            if (source is Iterator<TSource> iterator)
```

Comment:
IIUC this change is removing the `GetCount()` method from unoptimized builds. Isn't this going to regress performance in certain scenaria?

Reply:
It's not. The interface was there and being tested for, but nothing implemented it, so the test would always be false.